src/devices/vdml_usd.cpp

```diff
@@ -21,5 +21,77 @@ std::int32_t is_installed(void) {
 	return usd_is_installed();
 }
 
+std::int32_t list_files_raw(const char* path, char* buffer, int32_t len) {
+	return usd_list_files_raw(path, buffer, len);
+}
+
+std::vector<std::string> list_files(const char* path) {
+	std::vector<std::string> files = {};
+	// malloc buffer to store file names
+	size_t buffer_size = 10000;
+	char *buffer = (char *) malloc(buffer_size);
```

Comment:
You should use the C++ `new char[buffer_size];`instead of C's `malloc` (and later `delete`instead of `free).

It's a bit less scary type-wise and very widely considered better stylistically for C++.

Reply:
Actually even better would be `std::unique_ptr<char[]> buffer = std::make_shared<char[]>(buffer_size)`and then passing `buffer.get()` to the C function. Then the `delete`happens automatically when you exit the function scope.